src/client/app/strings/index.ts

```diff
@@ -50,7 +50,17 @@ export function Strings(): ClientStrings {
 }
 
 export function getDisplayLanguage() {
-    return (getRawDisplayLanguage() || 'en').toLowerCase();
+    let rawDisplayLanguage = getRawDisplayLanguage();
+    if (rawDisplayLanguage) {
+        rawDisplayLanguage = rawDisplayLanguage.toLowerCase();
+        for (let language of availableLanguages) {
+            if (rawDisplayLanguage === language.value) {
+                return rawDisplayLanguage;
+            }
+        }
+    }
+
+    return (rawDisplayLanguage || 'en').substr(0, 2);
```

Comment:
@dsaunders45459 , once you've taken a look at the above comment, feel free to check in as is if it's not a concern.  I wasn't sure if it's important or not in light of the new Chinese strings.